pkg/component/dependencywatchdog/access.go

```diff
@@ -67,42 +77,107 @@ type dependencyWatchdogAccess struct {
 
 // AccessValues contains configurations for the component.
 type AccessValues struct {
-	// ServerOutOfCluster is the out-of-cluster address of a kube-apiserver.
-	ServerOutOfCluster string
 	// ServerInCluster is the in-cluster address of a kube-apiserver.
 	ServerInCluster string
 }
 
+// DWDAccess is an interface to control the life cycle of the DWD object
+type DWDAccess interface {
```

Comment:
Typically, we simply call this `Interface`:

```suggestion
// Interface exposes methods for deploying dependency-watchdog.
type Interface interface {
```